### PR TITLE
Renamed SuperellipseShape to ContinuousCornerBorder

### PR DESCRIPTION
**Pull request by Salby**
When I initially was developing `SuperellipseShape` I changed the method of calculating the shape to allow more control over the individual corners. As a result of this, it is no longer a mathematical superellipse, and it is therefore misleading to call it that.

I've changed the name to ContinuousCornerBorder because it still implies the continuity of the curved corners compared to `RoundedRectangleBorder`.

I changed `Shape` to `Border` to be more in line with the other default shapes (`RoundedRectangleBorder` & `BeveledRectangleBorder`).

### CODE REVIEW

File: packages/flutter/lib/src/painting/continuous_corner_border.dart [hunk 1]
```diff
@@ -124,7 +128,8 @@ class SuperellipseShape extends ShapeBorder {
 
   @override
   Path getInnerPath(Rect rect, {TextDirection textDirection}) {
-    return _getPath(borderRadius.resolve(textDirection).toRRect(rect).deflate(side.width));
+    return _getPath(
+        borderRadius.resolve(textDirection).toRRect(rect).deflate(side.width));
```
Comment (Hixie) [inline packages/flutter/lib/src/painting/continuous_corner_border.dart]: nit: this was better as before

File: packages/flutter/lib/src/painting/continuous_corner_border.dart [hunk 2]
```diff
@@ -151,9 +156,8 @@ class SuperellipseShape extends ShapeBorder {
   bool operator ==(dynamic other) {
     if (runtimeType != other.runtimeType)
       return false;
-    final SuperellipseShape typedOther = other;
-    return side == typedOther.side
-        && borderRadius == typedOther.borderRadius;
+    final ContinuousCornerBorder typedOther = other;
+    return side == typedOther.side && borderRadius == typedOther.borderRadius;
```
Comment (Hixie) [inline packages/flutter/lib/src/painting/continuous_corner_border.dart]: nit: this was better on two lines as before

**Comment by Hixie** [comment]
There's at least one other place in the API docs that mentions the old class name that should also get updated.

**Review by jslavitz — COMMENTED** [review]
Hey Salby! Not sure if you saw my PR: https://github.com/flutter/flutter/pull/27523. We actually ended up finding a different equation/ set of curves that builds this shape in a more suitable way. As such, we intend to revert your PR very soon and so we will be closing this PR. However, we really appreciate your contribution (you can see some of the code has made it into the new PR, as well as this marvelous naming idea!). We are really sorry to have to revert your code, but we would very much appreciate future contributions! Thanks again!

**Pull request closed without merging**